File: eAnimalShelter/eAnimalShelter.Services/Database/Donation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using eAnimalShelter.Model.Enums;

namespace eAnimalShelter.Services.Database
{
    public class Donation
    {
        [Key]
        public int DonationId { get; set; }
        public int UserId { get; set; }
         [ForeignKey("UserId")]
        public User User { get; set; } = null!;
        public decimal Amount { get; set; }
        public DateTime? DonationDate { get; set; }
        public string PaymentMethod { get; set; } = string.Empty;
        public string? StripePaymentIntentId { get; set; }
        public DonationStatus TransactionStatus { get; set; }
        public string? Note { get; set; }
        public string? ReceiptPdfPath { get; set; }
    }

    
}

File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/donation.dart
class Donation {
  int? donationId;
  int? userId;

  String? userFullName;

  double? amount;

  DateTime? donationDate;

  String? paymentMethod;

  String? stripePaymentIntentId;

  int transactionStatus;

  String? note;

  String? receiptPdfPath;

  Donation({
    this.donationId,
    this.userId,
    this.userFullName,
    this.amount,
    this.donationDate,
    this.paymentMethod,
    this.stripePaymentIntentId,
    this.transactionStatus = 2,
    this.note,
    this.receiptPdfPath,
  });

  factory Donation.fromJson(
    Map<String, dynamic> json,
  ) {
    return Donation(
      donationId: json["donationId"],
      userId: json["userId"],
      userFullName: json["userFullName"],
      amount: (json["amount"] as num?)?.toDouble(),
      donationDate: json["donationDate"] != null
          ? DateTime.parse(json["donationDate"])
          : null,
      paymentMethod: json["paymentMethod"],
      stripePaymentIntentId: json["transactionId"],
      transactionStatus:
          json["transactionStatus"] ?? 0,
      note: json["note"],
      receiptPdfPath:
          json["receiptPdfPath"],
    );
  }
}